roles/ocp_on_libvirt/tasks/libvirt_host_destroy.yml

```diff
@@ -1,20 +1,31 @@
 ---
-- name: "Does node exist"
+- name: "Does node exist - {{ host['name'] }}"
+  community.libvirt.virt:
+    command: list_vms
+  register: _ool_vms_list
+
+- name: "Set node_not_exists"
+  ansible.builtin.set_fact:
+    ool_node_exists: "{{ host['name'] in _ool_vms_list.list_vms }}"
+
+- name: "Get status if exists - {{ host['name'] }}"
   community.libvirt.virt:
     name: "{{ host['name'] }}"
     command: status
-  ignore_errors: true
-  register: node_exists
+  register: _ool_node_status
+  when: ool_node_exists
 
-- name: "Destroy node"
+- name: "Destroy {{ host.name }}"
   community.libvirt.virt:
     name: "{{ host['name'] }}"
     command: destroy
-  when: node_exists['status'] is defined and node_exists['status'] == 'running'
+  when: _ool_node_status['status'] is defined and _ool_node_status['status'] == 'running'
 
-- name: "Undefine node"
+- name: "Undefine {{ host.name }}"
   become: true
-  ansible.builtin.command:
-    # community.libvirt.virt undefine can't specify --nvram
-    cmd: "virsh undefine --remove-all-storage --nvram {{ host['name'] }}"
-  when: not node_exists.failed
+  community.libvirt.virt:
+    name: "{{ host['name'] }}"
+    command: undefine
+    flags: nvram
+  when: ool_node_exists
```

Comment:
Should we use the same condition than in the previous task? Just to follow the same terminology: `when: _ool_node_status['status'] is defined and _ool_node_status['status'] == 'running'`

Reply:
No need cause the previous one should move the status from running to shutdown

Reply:
ack